DataFormats/Portable/interface/PortableCollection.h

```diff
@@ -94,4 +97,47 @@ namespace cms::alpakatools {
   };
 }  // namespace cms::alpakatools
 
+// Specialize the MemcpyTraits for PortableColletion
+namespace edm {
+
+  template <typename T>
+  struct MemcpyTraits<PortableHostCollection<T>> {
+    static std::vector<std::size_t> properties(PortableHostCollection<T> const& object) {
+      return {object->metadata().size()};
+    }
+
+    static std::vector<std::pair<void*, std::size_t>> regions(PortableHostCollection<T>& object) {
+      void* address = object.buffer().data();
+      size_t size = alpaka::getExtentProduct(object.buffer());
+      return {{address, size}};
+    }
+
+    static std::vector<std::pair<void const*, std::size_t>> regions(PortableHostCollection<T> const& object) {
+      const void* address = object.buffer().data();
+      size_t size = alpaka::getExtentProduct(object.buffer());
+      return {{address, size}};
+    }
+  };
+
+  template <typename T, typename TDev>
+  struct MemcpyTraits<PortableDeviceCollection<T, TDev>> {
+    static std::vector<std::size_t> properties(PortableDeviceCollection<T, TDev> const& object) {
+      return {object->metadata().size()};
+    }
+
+    static std::vector<std::pair<void*, std::size_t>> regions(PortableDeviceCollection<T, TDev>& object) {
```

Comment:
To make the specialisation for `PortableDeviceCollection` work, some kind of pass-through specialisation for `edm::DeviceProduct` is still needed.